android/app/src/main/res/layout/max_widget_layout.xml

```diff
@@ -0,0 +1,427 @@
+        android:orientation="vertical">
+
+        <LinearLayout
+            android:id="@+id/locationLayout"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:orientation="horizontal">
+
+            <TextView
+                android:id="@+id/locationNameTextView"
+                style="@style/locationName"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_horizontal"
+                android:gravity="center_horizontal"
+                tools:text="Helsinki," />
+
+            <TextView
+                android:id="@+id/locationRegionTextView"
+                style="@style/locationRegion"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_horizontal"
+                android:gravity="center_horizontal"
+                android:textSize="15sp"
+                tools:text="Suomi" />
+
+        </LinearLayout>
+
+        <TextView
+            android:id="@+id/timeTextView"
+            style="@style/currentWeatherTime"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:gravity="center_horizontal"
+            android:textSize="15sp"
+            tools:text="klo 11:00" />
+
+        <LinearLayout
+            android:id="@+id/weatherLayout"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_horizontal"
+            android:layout_marginTop="3dp"
+            android:layout_marginBottom="7dp"
+            android:orientation="horizontal">
+
+            <ImageView
+                android:id="@+id/weatherIconImageView"
+                android:layout_width="46dp"
+                android:layout_height="46dp"
+                android:layout_marginEnd="18dp"
+                android:contentDescription="TODO"
+                tools:src="@drawable/s32_light" />
+
+            <TextView
+                android:id="@+id/temperatureTextView"
+                style="@style/temperature"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_vertical"
+                tools:text="+12" />
+
+            <TextView
+                android:id="@+id/temperatureUnitTextView"
+                style="@style/temperatureUnit"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="6dp"
+                tools:text="°C" />
+
+        </LinearLayout>
+
+        <RelativeLayout
+            android:id="@+id/hourForecastRowLayout"
```

Comment:
All forecast rows have too little padding compared to design (rows are vertically too thin).

Reply:
Done